File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpContainerTakeData.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RMcpContainerTakeData(
        boolean dryRun,
        int requestedCount,
        int movedCount,
        RMcpContainerData.Slot movedItem,
        RMcpContainerMoveData.Endpoint from,
        Integer toInventorySlot,
        List<Integer> targetInventorySlots
) {
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpContainerTakeData.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpContainerTakeData(
    val dryRun: Boolean,
    val requestedCount: Int,
    val movedCount: Int,
    val movedItem: RMcpContainerData.Slot,
    val from: RMcpContainerMoveData.Endpoint,
    val toInventorySlot: Int,
    val targetInventorySlots: List<Int>
)
